### ISSUE TITLE
Internal Server Error

### ISSUE THREAD

**Issue by junpet** [te-online/timemanager#100]
# Description

After installation clicking on the toolbar icon an internal server error happens. Let me know if you need more information.

# Errors

I found these errors in the log. I censored some sensitive information.

```
[index] Error: Doctrine\DBAL\Exception\SyntaxErrorException: An exception occurred while executing a query: SQLSTATE[42601]: Syntax error: 7 ERROR:  trailing junk after parameter at or near "$3O"
LINE 1: ...id" = $2) AND "oc_timemanager_client"."status" != $3ORDER BY...
                                                             ^ at <<closure>>

 0. /var/www/html/3rdparty/doctrine/dbal/src/Connection.php line 1780
    Doctrine\DBAL\Driver\API\PostgreSQL\ExceptionConverter->convert(Doctrine\DBAL\Driver\PDO\Exception {}, Doctrine\DBAL\Query {})
 1. /var/www/html/3rdparty/doctrine/dbal/src/Connection.php line 1719
    Doctrine\DBAL\Connection->handleDriverException(Doctrine\DBAL\Driver\PDO\Exception {}, Doctrine\DBAL\Query {})
 2. /var/www/html/3rdparty/doctrine/dbal/src/Statement.php line 193
    Doctrine\DBAL\Connection->convertExceptionDuringQuery(Doctrine\DBAL\Driver\PDO\Exception {}, "SELECT DISTINCT ... ;", ["censored","censored","deleted"], [2,2,2])
 3. /var/www/html/lib/private/DB/PreparedStatement.php line 87
    Doctrine\DBAL\Statement->execute(null)
 4. /var/www/html/lib/public/AppFramework/Db/Mapper.php line 252
    OC\DB\PreparedStatement->execute()
 5. /var/www/html/lib/public/AppFramework/Db/Mapper.php line 340
    OCP\AppFramework\Db\Mapper->execute("SELECT DISTINCT ... ;", ["censored","censored","deleted"], null, null)
 6. /var/www/html/custom_apps/timemanager/db/objectmapper.php line 264
    OCP\AppFramework\Db\Mapper->findEntities("SELECT DISTINCT ... ;", ["censored","censored","deleted"])
 7. /var/www/html/custom_apps/timemanager/controller/pagecontroller.php line 100
    OCA\TimeManager\Db\ObjectMapper->findActiveForCurrentUser("name", true)
 8. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 225
    OCA\TimeManager\Controller\PageController->index("")
 9. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 133
    OC\AppFramework\Http\Dispatcher->executeController(OCA\TimeManager\ ... {}, "index")
10. /var/www/html/lib/private/AppFramework/App.php line 172
    OC\AppFramework\Http\Dispatcher->dispatch(OCA\TimeManager\ ... {}, "index")
11. /var/www/html/lib/private/Route/Router.php line 298
    OC\AppFramework\App::main("OCA\\Timemanage ... r", "index", OC\AppFramework\ ... {}, ["timemanager.page.index"])
12. /var/www/html/lib/base.php line 1047
    OC\Route\Router->match("/apps/timemanager/")
13. /var/www/html/index.php line 36
    OC::handleRequest()

GET /apps/timemanager/
from censored by censored at 2022-11-07T06:33:05+00:00
```

```
[PHP] Error: Error: Return type of OCA\IssueTemplate\Section::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice at /var/www/html/custom_apps/issuetemplate/lib/Section.php#80 at <<closure>>

 0. /var/www/html/custom_apps/issuetemplate/lib/Section.php line 26
    OC\Log\ErrorHandler::onError(8192, "Return type of  ... e", "/var/www/html/c ... p", 80)
 1. /var/www/html/lib/composer/composer/ClassLoader.php line 571
    include("/var/www/html/c ... p")
 2. /var/www/html/lib/composer/composer/ClassLoader.php line 428
    Composer\Autoload\includeFile("/var/www/html/c ... p")
 3. /var/www/html/custom_apps/issuetemplate/lib/Sections/LogSection.php line 29
    Composer\Autoload\ClassLoader->loadClass("OCA\\IssueTemplate\\Section")
 4. /var/www/html/lib/composer/composer/ClassLoader.php line 571
    include("/var/www/html/c ... p")
 5. /var/www/html/lib/composer/composer/ClassLoader.php line 428
    Composer\Autoload\includeFile("/var/www/html/c ... p")
 6. <<closure>>
    Composer\Autoload\ClassLoader->loadClass("OCA\\IssueTempl ... n")
 7. /var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php line 114
    ReflectionClass->__construct("OCA\\IssueTempl ... n")
 8. /var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php line 133
    OC\AppFramework\Utility\SimpleContainer->resolve("OCA\\IssueTempl ... n")
 9. /var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php line 465
    OC\AppFramework\Utility\SimpleContainer->query("OCA\\IssueTempl ... n")
10. /var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php line 437
    OC\AppFramework\DependencyInjection\DIContainer->queryNoFallback("OCA\\IssueTempl ... n")
11. /var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php line 89
    OC\AppFramework\DependencyInjection\DIContainer->query("OCA\\IssueTempl ... n", true)
12. <<closure>>
    OC\AppFramework\Utility\SimpleContainer->OC\AppFramework\Utility\{closure}("*** sensitive parameters replaced ***")
13. /var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php line 108
    array_map(Closure {}, [ReflectionParam ... }])
14. /var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php line 116
    OC\AppFramework\Utility\SimpleContainer->buildClass(ReflectionClass  ... "})
15. /var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php line 133
    OC\AppFramework\Utility\SimpleContainer->resolve("OCA\\IssueTemplate\\Settings\\Admin")
16. /var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php line 465
    OC\AppFramework\Utility\SimpleContainer->query("OCA\\IssueTemplate\\Settings\\Admin")
17. /var/www/html/lib/private/ServerContainer.php line 145
    OC\AppFramework\DependencyInjection\DIContainer->queryNoFallback("OCA\\IssueTemplate\\Settings\\Admin")
18. /var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php line 57
    OC\ServerContainer->query("OCA\\IssueTemplate\\Settings\\Admin")
19. /var/www/html/lib/private/Settings/Manager.php line 206
    OC\AppFramework\Utility\SimpleContainer->get("OCA\\IssueTemplate\\Settings\\Admin")
20. /var/www/html/lib/private/Settings/Manager.php line 351
    OC\Settings\Manager->getSettings("admin", "overview")
21. /var/www/html/apps/settings/lib/Controller/CommonSettingsTrait.php line 86
    OC\Settings\Manager->getAllowedAdminSettings("overview", OC\User\User {})
22. /var/www/html/apps/settings/lib/Controller/CommonSettingsTrait.php line 117
    OCA\Settings\Controller\AdminSettingsController->formatSections([[OCA\Settings\S ... "], "overview", "admin", "admin")
23. /var/www/html/apps/settings/lib/Controller/CommonSettingsTrait.php line 67
    OCA\Settings\Controller\AdminSettingsController->formatAdminSections("admin", "overview")
24. /var/www/html/apps/settings/lib/Controller/CommonSettingsTrait.php line 148
    OCA\Settings\Controller\AdminSettingsController->getNavigationParameters("admin", "overview")
25. /var/www/html/apps/settings/lib/Controller/AdminSettingsController.php line 68
    OCA\Settings\Controller\AdminSettingsController->getIndexResponse("admin", "overview")
26. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 225
    OCA\Settings\Controller\AdminSettingsController->index("overview")
27. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 133
    OC\AppFramework\Http\Dispatcher->executeController(OCA\Settings\Con ... {}, "index")
28. /var/www/html/lib/private/AppFramework/App.php line 172
    OC\AppFramework\Http\Dispatcher->dispatch(OCA\Settings\Con ... {}, "index")
29. /var/www/html/lib/private/Route/Router.php line 298
    OC\AppFramework\App::main("OCA\\Settings\\ ... r", "index", OC\AppFramework\ ... {}, ["overview","set ... "])
30. /var/www/html/lib/base.php line 1047
    OC\Route\Router->match("/settings/admin/overview")
31. /var/www/html/index.php line 36
    OC::handleRequest()

GET /settings/admin/overview
from censored by censored at 2022-11-07T06:33:57+00:00
```

# Deployment

I wanted to use Issue reporting, but it does not work for some reason. Now, I don't have time, let me know if you need more information.

Nextcloud: v25.0.1 (in Docker, fpm with Nginx and Traefik)
TimeManager: v0.3.4

**Comment by te-online** [comment]
@junpet Thanks for reporting! 👍 

I need to refactor all these ugly queries to use query builder, but haven't gotten around to it yet...

It looks to me like in [`db/objectmapper.php:260`](https://github.com/te-online/timemanager/blob/main/db/objectmapper.php#L260) a space is missing at the end of the first part of the statement.

Some databases may be picky about this missing space...

Do you have the ability to test a patch? 🤓 

If you replace the line like this, does it still throw the error
```diff
- "WHERE (`*PREFIX*timemanager_share`.`recipient_user_id` = ? OR `$this->tableName`.`user_id` = ?) AND `$this->tableName`.`status` != ?" .
+ "WHERE (`*PREFIX*timemanager_share`.`recipient_user_id` = ? OR `$this->tableName`.`user_id` = ?) AND `$this->tableName`.`status` != ? " .
```

---

About the issue template: I had no idea a feature like this existed 😅 But it looks to me more like this is broken in [nextcloud/server](https://github.com/nextcloud/server) ...

**Comment by junpet** [comment]
Well, I forgot to mention I use PostgreSQL v15.0.0 as the backend. Sorry for that, I was in a hurry, I should have to think about that because of the SQL error. 

I've edited the file you mentioned and now it's working. Thank you very much! :)

**Comment by te-online** [comment]
Should be fixed in 0.3.5
